pkg/handler/server_ssh.go

```diff
@@ -783,10 +806,32 @@ func (s *Server) buildSSHClient(tokenInfo *model.ConnectToken) (*srvconn.SSHClie
 		gateways = []model.Gateway{*tokenInfo.Gateway}
 	}
 	sshAuthOpts := buildSSHClientOptions(&asset, &account, gateways)
+	// add reuse ssh client
+	enableReused := config.GetConf().ReuseConnection
+	reusedKey := GenerateSSHTokenResueKey(tokenInfo)
+	if enableReused {
+		if client, ok := srvconn.GetClientFromCache(reusedKey); ok {
+			logger.Infof("Reused ssh client key: %s", reusedKey)
+			return client, nil
+		}
+	}
 	sshClient, err := srvconn.NewSSHClient(sshAuthOpts...)
 	if err != nil {
 		logger.Errorf("Get SSH Client failed: %s", err)
 		return sshClient, err
 	}
+	if enableReused {
+		srvconn.AddClientCache(reusedKey, sshClient)
+	}
 	return sshClient, nil
 }
+
+func GenerateSSHTokenResueKey(tokenInfo *model.ConnectToken) string {
+	userId := tokenInfo.User.ID
+	assetId := tokenInfo.Asset.ID
+	ip := tokenInfo.Asset.Address
+	port := tokenInfo.Asset.ProtocolPort("ssh")
+	accountUsername := tokenInfo.Account.Username
+	return fmt.Sprintf("SSHD_%s_%s_%s_%d_%s",
+		userId, assetId, ip, port, accountUsername)
+}
```

Comment:
There is no syntax checking or type assertion provided here for you to point out. Please provide proper Go code with its intended functionality so I can check if there are any irregularities, potential issues, or optimization suggestions that need your attention please.